mission-control-client/src/components/dashboard/products/ProductList.js

```diff
@@ -3,22 +3,47 @@ import SearchIcon from "@material-ui/icons/Search";
 import Product from "./Product";
 import { connect } from "react-redux";
 import { setActiveProduct } from "../../../actions/activeProductActions";
+import { useHistory, useLocation } from 'react-router-dom' 
+import { set } from "react-ga";
 
 const ProductList = props => {
+
+  // getting the current location and splitting into array to check for current location later
+  const location = useLocation().pathname.split('/')
+
+  const history = useHistory()
+
   useEffect(() => {
-    setFiltered({ products: props.products });
 
-    if (filtered.products.length > 0) {
+    setFiltered({ products: props.products });
+    // if the user has matching products in the search bar and if there isn't already a product currently active
+    if (filtered.products.length > 0 && !props.activeProductStore.active) {
       props.setActiveProduct(filtered.products[0]);
-    } else {
+
+    // if there is already an active product in the store render that product until user has clicked on that card
+    // ! important for use on any other route other then the default dashboard
+    } else if (props.activeProductStore.active){
+      return
+
+    //if the user hasn't selected any current product render the first product by default
+    }else{
       props.setActiveProduct(props.products[0]);
     }
-  }, [props.products]);
+
+  }, [props.products, props.activeProductStore.active]);
 
   const [filtered, setFiltered] = useState({ products: [] });
 
   const setProductHandler = el => {
-    props.setActiveProduct(el);
```

Comment:
This is something we definitely need? Were we getting errors with a product being selected and the user not being on the dashboard? @armandoroman1016